rclcpp/include/rclcpp/service.hpp

```diff
@@ -66,10 +67,9 @@ class ServiceBase
 protected:
   RCLCPP_DISABLE_COPY(ServiceBase);
 
-  std::shared_ptr<rcl_node_t> node_handle_;
+  rcl_node_t * node_handle_;
```

Comment:
The second constructor takes raw `rcl` pointer. I am feeling a bit concerned about turning a raw pointer in a shared pointer since we'd miss 1 ref count. 
But generally, do we need this pointer as a shared component or would a `unique_ptr` help here as the class anyways is marked as non-copyable.